dss/dynamodb/__init__.py

```diff
@@ -24,18 +29,23 @@ def put_item(*, table: str, hash_key: str, sort_key: str=None, value: str, overw
     :param str hash_key: 1st primary key that can be used to fetch associated sort_keys and values.
     :param str sort_key: 2nd primary key, used with hash_key to fetch a specific value.
                          Note: If not specified, this will PUT only 1 key (hash_key) and 1 value.
-    :param str overwrite: Don't overwrite if this parameter exists.  For example, setting this
-                          to 'sort_key' won't overwrite if that sort_key already exists in the table.
+    :param str dont_overwrite: Don't overwrite if this parameter exists.  For example, setting this
+                               to 'sort_key' won't overwrite if that sort_key already exists in the table.
     :return: None
     """
     query = {'TableName': table,
              'Item': _format_item(hash_key=hash_key, sort_key=sort_key, value=value)}
-    if overwrite:
-        query['ConditionExpression'] = f'attribute_not_exists({overwrite})'
-    db.put_item(**query)
+    if dont_overwrite:
+        query['ConditionExpression'] = f'attribute_not_exists({dont_overwrite})'
 
+    try:
+        db.put_item(**query)
+    except ClientError as e:
+        if e.response['Error']['Code'] != 'ConditionalCheckFailedException':
+            raise
```

Comment:
I feel like encountering an existing item while `dont_overwrite != None` should raise.

If I were writing client methods, I'd want to be able to react when a DDB operation is not idempotent. What do you think?

Reply:
I think you're right.  Will put into the collection method.